File: labs/J30_Visitor/python/complete/filesysteminfo.py
class FileSystemInfo():
    name = ""

    def __init__(self, name):
        self.name = name


class FileInfo(FileSystemInfo):
    def accept(self, visitor):
        visitor.visit_file(self)


class DirectoryInfo(FileSystemInfo):
    contents = []

    def __init__(self, name):
        self.name = name
        self.contents=[]

    def getContents(self):
        return self.contents

    def addChild(self, child):
        self.contents.append(child)

    def accept(self, visitor):
        visitor.visit_directory(self)

File: labs/J30_Visitor/java/complete/app/src/main/java/VisitorComplete/FileSystemInfo.java
package VisitorComplete;

public abstract class FileSystemInfo {
  private String name="";

  public String getName() {
    return name;
  }

  public void setName(String name) {
    this.name = name;
  }

  public FileSystemInfo(String name) {
    this.name = name;
  }

  public abstract void accept(IFileSystemVisitor visitor);

}

